### ISSUE TITLE
Accessibility feedback for currency symbol addition to va-number-input from DST

### ISSUE THREAD

**Issue by jamigibbs** [department-of-veterans-affairs/va.gov-team#55335]
## What does your team need support for? Check all that apply.
- [ ] Launching one or more unauthenticated pages, online forms, apps, or tools on VA.gov
- [ ] Making revisions to one or more existing VA.gov pages
- [X] Something else 
 
We are adding a currency symbol to the `va-number-input` web component and we will need some accessibility feedback on how we can best provide accessibility support for it:

![Screenshot 2023-03-17 at 11 07 36 AM](https://user-images.githubusercontent.com/872479/225958456-44bdf77e-5854-44c7-8384-a91e190c6d5c.png)

## Will this new product be released incrementally (for instance 25% of users initially)? 
  - [ ] Yes
  - [x] No
  
## Supporting artifacts

- The original ticket requesting this addition: 
https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1280

As discussed in the design system council:

> DK mentioned the aria-label and SaraK commented that VoiceOver announces currency but other screen readers may not. Sarah can guide our implementation should we get to it.

- Here is the active PR:
 https://github.com/department-of-veterans-affairs/component-library/pull/651

- A live example of the currency addition can be previewed here: 
https://1280-update-va-number-input--60f9b557105290003b387cd5.chromatic.com/?path=/docs/components-va-number-input--with-currency

- Here is a relevant Slack convo:
https://dsva.slack.com/archives/C8E985R32/p1679005727885479

## Will this work be going through the Collaboration Cycle?
- [ ] Yes 
- [x] No

## When does this work need to be done?
- Estimated launch date: 
- Estimated staging review date:
- Content and IA work needed by: 

## Do you plan to bring this to an upcoming content office hours session?
- [ ] Yes, benefit content office hours (Thursdays, 3:00 p.m. to 3:45 p.m. ET)
- [ ] Yes, health  content office hours (Thursdays, 11:00 a.m. to 11:30 a.m. ET)
- [ ] Yes, unauth office hours (Mondays, 11:00 a.m. to 11:30 a.m. ET)
- [ ] No, but I'd like to schedule time to talk about this request
- [x] No, let's work asynchronously and meet if needed

**Note:** If we think this work would benefit from a collaborative session with you, we may ask you to bring it to office hours or set up a separate time to meet.

## About your team

- Team name: VA Design System
-	OCTO-DE product owner: Matthew Dingee
-	Product manager: Carol Wong
-	Designer: N/A
-	FE engineer: Jami Gibbs
-	Product/team Slack channel: `#platform-design-system`

## Next steps

**Once you’ve submitted this ticket, please post a link to this issue in the [#sitewide-content-ia](https://dsva.slack.com/channels/sitewide-content-ia) Slack channel and tag Randi Hecht.**

If you also need engineering support from the Public Websites team, fill out their [intake request form](https://github.com/department-of-veterans-affairs/va.gov-team/issues/new?assignees=Public+Websites%2C+Sitewide+content&labels=vsa-public-websites%2C+vsa%2C+vsa-public-websites-intake%2C+sitewide-content%2C+needs-grooming&template=public-websites-intake.md&title=%3CType+of+Request%3E+from+%3CTeam%3E).
  
If you need a page/URL redirected, a URL changed or a vanity URL set up, please submit a [Redirect, URL change, or vanity URL request](https://github.com/department-of-veterans-affairs/va.gov-team/issues/new?assignees=mnorthuis&labels=ia&template=redirect-request.md&title=Redirect+Request)

**Comment by jamigibbs** [comment]
Other relevant work that is needing the currency symbol added to the va-number-input component: 
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/55213

**Comment by SarahKay8** [comment]
@Dr-Pongo Hey Andrew. I have this marked as a 4. Will try to get this done before end of our current sprint.

**Comment by SarahKay8** [comment]
@jamigibbs so This might be something where having additional context on the label for everyone could be helpful. For example, adding the currency, like "My input (in USD)".. It could also be worthwhile to indicate in help text to only enter whole numbers and add that with an `aria-describedby` reference, since it seems like  `va-number-input` doesn't allow partial dollar amounts.

**Comment by SarahKay8** [comment]
@jamigibbs 

If we don't include in context that its in dollars then I would go for something like this..



<label for="input">My input</label>

<div>
<span id="symbol">$</span>
<input id="input" aria-labelledby="input symbol" type="text" inputmode="numeric" pattern="[0-9]*">
</div>


aria-labelledby takes in an ID (or list of ID's) to produce an accessible name for an element. In this case, "input" refers to the element self, which is already labelled via the <label> element via the "for" attribute. "symbol" is the ID of the $. So together, the announcement for a screen reader would be along the lines of "My input dollar, edit" when using aria-labelledby="input symbol".

@jamigibbs

**Comment by SarahKay8** [comment]
@jamigibbs 

If we don't include in context that it's in dollars then I would go for something like this 


`<label for="input">My input</label>

<div>
<span id="symbol">$</span>
<input id="input" aria-labelledby="input symbol" type="text" inputmode="numeric" pattern="[0-9]*">
</div>
`

`aria-labelledby` takes in an ID (or list of ID's) to produce an accessible name for an element. In this case, "input" refers to the element self, which is already labelled via the <label> element via the "for" attribute. "symbol" is the ID of the $. So together, the announcement for a screen reader would be along the lines of "My input dollar, edit" when using aria-labelledby="input symbol".

**Comment by SarahKay8** [comment]
If we plan on providing context in the label than yes that markup is sufficient. @jamigibbs 




If we decide not to provide context in the label i would go for something like this:

<label for="input">My input</label>

< div >

< span id="symbol"> $ </span >

< input id="input" aria-labelledby="input symbol" type="text" inputmode="numeric" pattern="[0-9]*" >

< /div >





aria-labelledby takes in an ID (or list of ID's) to produce an accessible name for an element. In this case, "input" refers to the element self, which is already labelled via the <label> element via the "for" attribute. "symbol" is the ID of the $. So together, the announcement for a screen reader would be along the lines of "My input dollar, edit" when using aria-labelledby="input symbol"


However, I like the first approach better. Just providing an option. lol

**Comment by jamigibbs** [comment]
Thanks @SarahKay8! I think providing an additional context with the symbol makes sense especially since the design system can't necessarily enforce the text being entered into the label in this way (I will try to add this as our recommendation to our documentation though).

**Comment by SarahKay8** [comment]
Sounds good! 🎉 @jamigibbs

**Comment by coforma-terry** [comment]
Going to close this one out - thank you all :)
